modules/internal/CPtr.chpl

```diff
@@ -223,8 +223,26 @@ module CPtr {
     :arg arr: the array for which we should retrieve a pointer
     :returns: a pointer to the array data
   */
-  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) {
-    if !arr._value.oneDData then halt("error: c_ptrTo(multi_ddata array");
+  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) && !arr._value.isDefaultRectangular() {
+    if !chpl__getActualArray(arr).oneDData then halt("error: c_ptrTo(multi_ddata array");
+    return c_pointer_return(arr[arr.domain.low]);
+  }
+
+  pragma "no doc"
+  inline proc c_ptrTo(arr: []) where arr._value.isDefaultRectangular() {
+    const val = arr._value;
+    if chpl__isArrayView(val) {
```

Comment:
Currently we really only need to specialize when checking if the array can be expressed as a single _ddata. They can all return the right thing by doing this:
```chpl
return c_pointer_return(arr[arr.domain.low]);
```

It may also be somewhat inconvenient to implement this as a method. For example, for DefaultRectangular the ArrayView already stores the necessary info to check whether it can be expressed as a single ddata. If we wrote this as a method, we'd have to do something like:
```chpl
chpl__getActualArray(arr).dsiToCPtr(chpl__getViewDom(arr));
```
And potentially recompute all the metadata within that method. What do you think?